editorial pass of §5 up to 5.3.2 (#196)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

## Appraisal Context initialisation {#sec-app-ctx-init}

During the initialization phase, the CoRIM Appraisal Context is loaded with various objects such as CoMID tags, CoSWID tags, and cryptographic validation key material (including raw public keys, root certificates, and intermediate CA certificate chains).
These objects will be utilized in the Evidence Appraisal phase that follows.
The primary goal of this phase is to ensure that all necessary information is available for subsequent processing.

### CoRIM Selection

All available CoRIMs are collected.

CoRIMs that are not within their validity period, or that cannot be associated with an authenticated and authorised source MUST be discarded.

Any CoRIM that has been secured by a cryptographic mechanism, such as a signature, that fails validation MUST be discarded.

Other selection criteria MAY be applied.
For example, if the Evidence format is known in advance, CoRIMs using a profile that is not understood by a Verifier can be readily discarded.

The selection process MUST yield at least one usable tag.

> Later stages will further select the CoRIMs appropriate to the Evidence Appraisal stage. TBC(tho)

Author: ```suggestion
Later stages will further select the CoRIMs appropriate to the Evidence Appraisal stage.
```

After the Verifier has processed all CoBOMs it MUST discard any tags which have not been activated by a CoBOM.

### Tags Identification and Validation

The Verifier chooses tags - including Concise Module ID Tags (CoMID, {{sec-comid}}), Concise Software ID Tags (CoSWID, {{-coswid}}), and/or Concise Trust Anchor Stores (CoTS, {{-ta-store}}) - from the selected CoRIMs.

The Verifier MUST discard all tags which are not syntactically and semantically valid.
In particular, any cross-referenced triples (e.g., CoMID-CoSWID linking triples) MUST be successfully resolved.

### Appraisal Context Construction

All of the validated and potentially useful tags are loaded into the Appraisal Context.
This concludes the Appraisal Context initialisation phase.

[^issue]: Content missing. Tracked at https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/96

## Evidence Collection {#sec-ev-coll}

During the Evidence collection phase, the Verifier communicates with attesters to gather Evidence.
The first part of this phase does not involve any cryptographic validation.

> This means that Verifiers can use untrusted code for their initial Evidence collection.

Reviewer: ```suggestion
This means that Verifiers MAY use untrusted code for their initial Evidence collection.
```

Author: RFC2119 terms are for interop requirements.  This one is an implementation choice instead.

The Evidence collection generates protocol-specific data and transcripts that are used as input to appraisal by the Verifier.

## Evidence Appraisal {#sec-ev-appraisal}

### Cryptographic validation of Evidence {#sec-crypto-validate-evidence}

If Evidence is cryptographically signed, its validation is the first step in Appraisal.

The way cryptographic signature validation works depends on the specific Evidence collection protocol being used.

> For instance, in DICE, a proof of liveness is carried out on the final key in the certificate chain.

Reviewer: ```suggestion
For instance, in DICE, a proof of liveness is carried out on the final key in the certificate chain (a.k.a., the alias certificate).
```

If this is successful, a suitable certification path is looked up in the Appraisal Context, based on linking information obtained from the DeviceID certificate (see Section 9.2.1 of {{DICE.Layer}}).
If a trusted root certificate is found, the usual X.509 certificate validation is performed.

> On the other hand, in PSA, the verification public key is looked up in the appraisal context using the `ueid` claim found in the PSA claims-set (see {{Section 4.2.1 of -psa-token}}).

Reviewer: Should this text be moved to the PSA profile?

Author: I am not against dropping this.  However, it provides a concrete example of how different profiles may plug in their logic.

Reviewer: Lines 1485-1487 appear to apply to a DICE example. Line 1488 doesn't begin with enough descriptive text to highlight it as a second example, vs. an extension to a DICE example. As a second example, it needs more development to make it understandable. I didn't want to spend too much time in this document reviewing content that applies to a different audience. Possibly, a reference to the psa document is sufficient for an interested reader to go there to obtain the well-described example?

> On the other hand, in PSA, the verification public key is looked up in the appraisal context using the `ueid` claim found in the PSA claims-set (see {{Section 4.2.1 of -psa-token}}).

Author: ```suggestion
As a second example, in PSA {{-psa-token}} the verification public key is looked up in the appraisal context using the `ueid` claim found in the PSA claims-set.
```
